### PR TITLE
Consolidate HDFS file system docs

### PR DESCRIPTION
**Pull request by mosabua**
## Description

See commit message and content for details. 

## Additional context and related issues

Part of the whole object storage docs refactor. 

## Release notes

(✅) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: lib/trino-hdfs/src/main/java/io/trino/hdfs/HdfsConfig.java [hunk 1]
```diff
@@ -105,7 +105,7 @@ public boolean isNewFileInheritOwnership()
     }
 
     @Config("hive.fs.new-file-inherit-ownership")
-    @ConfigDescription("File system permissions for new directories")
+    @ConfigDescription("Flag to determine if new files inherit the ownership information from the directory.")
```
Comment (mosabua) [inline lib/trino-hdfs/src/main/java/io/trino/hdfs/HdfsConfig.java:108]: Not sure..

File: docs/src/main/sphinx/object-storage/file-system-hdfs.md [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+    ```text
+    hive.config.resources=/etc/hadoop/conf/core-site.xml
+    ```
+
+    Only specify additional configuration files if necessary for your setup, and
+    reduce the configuration files to have the minimum set of required
+    properties. Additional properties may cause problems.
+* - `hive.fs.new-directory-permissions`
+  - Controls the permissions set on new directories created for schemas and
+    tables. Value must either `skip` or an octal number, with a leading 0. If
```
Comment (electrum) [inline docs/src/main/sphinx/object-storage/file-system-hdfs.md:43]: "must either be"

File: docs/src/main/sphinx/object-storage/file-system-hdfs.md [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+* - `hive.dfs.connect.timeout`
+  - Timeout [duration](prop-type-duration) for connection operations to HDFS.
+    Defaults to `500ms`.
+* - `hive.dfs.connect.max-retries`
+  - Maximum number of retries for HDFS connection attempts. Defaults to `5`.
+* - `hive.dfs.key-provider.cache-ttl`
+  - Caching time [duration](prop-type-duration) for the key provider. Defaults
+    to `30min`.
+* - `hive.dfs.domain-socket-path`
+  - Path to the directory for the domain socket. The directory must exist on
```
Comment (electrum) [inline docs/src/main/sphinx/object-storage/file-system-hdfs.md:66]: > Path to the UNIX domain socket for the DataNode. This path must exist on each node. For example, `/var/lib/hadoop-hdfs/dn_socket`.

**Review by electrum — APPROVED** [review]

File: docs/src/main/sphinx/object-storage/file-system-hdfs.md [hunk 4]
```diff
@@ -0,0 +1,262 @@
+# HDFS file system support
+
+Trino includes support to access the [Hadoop Distributed File System
+(HDFS)](https://hadoop.apache.org/) with a catalog using the Delta Lake, Hive,
```
[the remark this answers is not in the document]
Comment (mosabua) [reply docs/src/main/sphinx/object-storage/file-system-hdfs.md:4]: Leaving in place .. its the same as in the other file system pages

File: docs/src/main/sphinx/object-storage/file-system-hdfs.md (hunk 2 above)
Comment (mosabua) [reply docs/src/main/sphinx/object-storage/file-system-hdfs.md:43]: Fixed

File: docs/src/main/sphinx/object-storage/file-system-hdfs.md (hunk 3 above)
Comment (mosabua) [reply docs/src/main/sphinx/object-storage/file-system-hdfs.md:66]: Updated

File: docs/src/main/sphinx/object-storage/file-system-hdfs.md [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+Hudi, or Iceberg connectors.
+
+Support is enabled by the default value `true` for the catalog property
```
[the remark this answers is not in the document]
Comment (mosabua) [reply docs/src/main/sphinx/object-storage/file-system-hdfs.md:7]: Updated .. with the mod to use "deactivated" ..

**Pull request merged**